src/plugins/controls/public/control_group/editor/control_editor.test.tsx

```diff
@@ -0,0 +1,251 @@
+import { act } from 'react-dom/test-utils';
+
+import { findTestSubject, mountWithIntl } from '@kbn/test-jest-helpers';
+import { stubDataView } from '@kbn/data-views-plugin/common/data_view.stub';
+
+import { ControlGroupInput } from '../types';
+import { pluginServices } from '../../services';
+import { ControlEditor, EditControlProps } from './control_editor';
+import { OptionsListEmbeddableFactory } from '../..';
+import {
+  DEFAULT_CONTROL_GROW,
+  DEFAULT_CONTROL_WIDTH,
+} from '../../../common/control_group/control_group_constants';
+import { mockControlGroupContainer, mockOptionsListEmbeddable } from '../../../common/mocks';
+import { RangeSliderEmbeddableFactory } from '../../range_slider';
+import { ControlGroupContainerContext } from '../embeddable/control_group_container';
+import {
+  OptionsListEmbeddableInput,
+  OPTIONS_LIST_CONTROL,
+  RANGE_SLIDER_CONTROL,
+} from '../../../common';
+
+describe('Data control editor', () => {
+  interface MountOptions {
+    componentOptions?: Partial<EditControlProps>;
+    explicitInput?: Partial<ControlGroupInput>;
+  }
+
+  pluginServices.getServices().dataViews.get = jest.fn().mockResolvedValue(stubDataView);
+  pluginServices.getServices().dataViews.getIdsWithTitle = jest
+    .fn()
+    .mockResolvedValue([{ id: stubDataView.id, title: stubDataView.getIndexPattern() }]);
+  pluginServices.getServices().controls.getControlTypes = jest
+    .fn()
+    .mockReturnValue([OPTIONS_LIST_CONTROL, RANGE_SLIDER_CONTROL]);
+  pluginServices.getServices().controls.getControlFactory = jest
+    .fn()
+    .mockImplementation((type: string) => {
+      if (type === OPTIONS_LIST_CONTROL) return new OptionsListEmbeddableFactory();
+      if (type === RANGE_SLIDER_CONTROL) return new RangeSliderEmbeddableFactory();
+    });
+
+  let controlEditor: ReactWrapper = new ReactWrapper(<></>);
+
+  async function mountComponent(options?: MountOptions) {
+    const controlGroupContainer = await mockControlGroupContainer(options?.explicitInput);
+
+    await act(async () => {
+      controlEditor = mountWithIntl(
+        <ControlGroupContainerContext.Provider value={controlGroupContainer}>
+          <ControlEditor
+            setLastUsedDataViewId={jest.fn()}
+            getRelevantDataViewId={() => stubDataView.id}
+            isCreate={true}
+            width={DEFAULT_CONTROL_WIDTH}
+            grow={DEFAULT_CONTROL_GROW}
+            onSave={jest.fn()}
+            onCancel={jest.fn()}
+            {...options?.componentOptions}
+          />
+        </ControlGroupContainerContext.Provider>
+      );
+    });
+    await new Promise(process.nextTick);
+    controlEditor.update();
+  }
+
+  const selectField = async (fieldName: string) => {
+    const option = findTestSubject(controlEditor, `field-picker-select-${fieldName}`);
+    await act(async () => {
+      option.simulate('click');
+    });
+    controlEditor.update();
+  };
+
+  describe.skip('creating a new control', () => {
```

Comment:
nit: why is this skipped?

Reply:
Woops - good catch. I had it skipped so that running the new tests would be quicker when I was writing them, but forgot to unskip it after I had them running 😆 Fixed in 1da21978f1dbd363a04cb5f0e83575571d0c97b7